src/pages/smart-contracts/stacks-api-client-tutorial.md

```diff
@@ -0,0 +1,727 @@
+
+```js
+const contractCallsOnly = r => {
+  console.log(r);
+  return r.tx_status === 'success' && r.tx_type === 'contract_call';
+};
+```
+
+and apply it to the response result:
+
+```js
+setRegistrations(response.results.filter(contractCallsOnly));
+```
+
+The final callback `fetchActivities` looks like this:
+
+```js
+const fetchActivities = useCallback(async () => {
+  // fetch activities
+  const response = await accountsApi.getAccountTransactions({
+    principal: `${CONTRACT_ADDRESS}.${CONTRACT_NAME}`,
+  });
+  console.log(response);
+  setRegistrations(response.results.filter(contractCallsOnly));
+}, []);
+```
+
+You should now see a list of hex strings representing the transactions.
+
+### 4. Step: Extract registry IDs
+
+The last step is to create a nice representation of the transactions. The `register` function returns the transaction id as a result. As mentioned above, it is an unsigned int (`uint`). The api call returns the transaction result as a hex string (`tx_result.hex`) like `0x0100000000000000000000000000000001`. It represents the serialized clarity value in hex encoding. The method `deserializeCV` creates a ClarityValue object out of the hex string, only the leading `0x` needs to be removed. The following line returns a `UintCV` object for the `register` transaction:
+
+```js
+deserializeCV(Buffer.from(activity.tx_result.hex.substr(2), 'hex'));
+```
+
+The client library defines types for all Clarity types. The corresponding type gives access to more properties. For example, `UintCV` has a `value` property for its number value:
+
+```js
+cv.value;
+```
+
+or you can use `cvToString` for an opinionated conversion to a string
+
+```js
+cvToString(cv);
+```
+
+### 5. Step: Display register details
+
+In addition to `tx_result`, the transaction object also contains a timestamp (`burn_block_time_iso`) and a `contract_call` object. The contract call object has properties like the function name (`contract_call.function_name`). Using this, the final representation for `register` transactions should contain the deserialized registry id and the timestamp. It could look like this:
+
+```js
+{
+  activities.map((activity, key) => {
+    if (activity.contract_call.function_name === 'register') {
+      const result = deserializeCV(Buffer.from(activity.tx_result.hex.substr(2), 'hex'));
+      return (
+        <React.Fragment key={key}>
+          Entry {result.value.toString()} was registered at {activity.burn_block_time_iso}.{' '}
+        </React.Fragment>
+      );
+    } else {
+      return null;
+    }
+  });
+}
+```
+
+-> Exercise: Show update transactions using the function arguments of the contract call `activity.contract_call.function_args`. This property is an array of clarity values. The `update` function takes two `buff` values.
+
+-> Note: The `AccountsApi` and other Api methods provide parameters to page through the results using `limit` and `offset`.
```

Comment:
Would link to the pagination chapter in the API references. Not sure if every developer knows what "paging through results" mean